## Ejemplo 1: Estimados de Locación



In [8]:
import pandas as pd

In [9]:
df = pd.read_csv('../../datasets/melbourne_housing-clean.csv', index_col=0)
df.head()

suburb           address  rooms type      price method seller_g  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S   Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S   Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP   Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI   Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB   Nelson   

        date  distance  postcode  bedroom_2  bathroom  car  land_size  \
0  3/12/2016       2.5    3067.0        2.0       1.0  1.0      202.0   
1  4/02/2016       2.5    3067.0        2.0       1.0  0.0      156.0   
2  4/03/2017       2.5    3067.0        3.0       2.0  0.0      134.0   
3  4/03/2017       2.5    3067.0        3.0       2.0  1.0       94.0   
4  4/06/2016       2.5    3067.0        3.0       1.0  2.0      120.0   

  council_area  latitude  longitude            region_name  property_count  
0        Yarra  -37.7996   144.9984  Northern Metropolitan          4019.0  
1        Yarra  -37.8079   144.9934  Northern Metropolitan          4019.0  
2        Yarra  -37.8093   144.9944  Northern Metropolitan          4019.0  
3        Yarra  -37.7969   144.9969  Northern Metropolitan          4019.0  
4        Yarra  -37.8072   144.9941  Northern Metropolitan          4019.0

### La media o promedio

La media o promedio se obtiene sumando todos los valores de un conjunto de datos numéricos y dividiéndolos entre la cantidad de valores que tenemos en nuestro conjunto.

Vamos a analizar la columna `price`. Veamos cuál es el "valor típico" obtenido usando la media (promedio):

In [11]:
df['price'].mean()

np.float64(1068141.895071269)

### La mediana

La mediana se obtiene de la siguiente manera:

1. Primero ordenamos de manera ascendente nuestros datos
2. Luego tomamos el valor que está justo en medio de nuestra secuencia ordenada de valores.
3. Si nuestro conjunto tiene un número par de valores y por lo tanto no tiene un valor justo a la mitad de la secuencia, se toma el promedio de los dos valores que se encuentran a la mitad de la secuencia.

Ahora veamos el "valor típico" obtenido usando la mediana:

In [12]:
df['price'].median()

np.float64(895500.0)

## Valores atípicos

Hemos aprendido a conseguir "valores típicos" se sirven para describir un conjunto de datos. Así como existen valores que se parecen a la norma, hay también valores que difieren mucho de la norma. Estos valores suelen sobresalir de nuestro conjunto de datos porque se encuentran a mucha distancia del grueso de las muestras.

Estos valores son los "valores atípicos". Hay veces en las que tener valores atípicos no nos preocupa, pero otras veces pueden ser peligrosos porque modifican nuestros estimados de locación, dándonos descripciones erróneas de nuestro conjunto de datos.

Más adelante aprenderemos a detectar valores atípicos, pero por lo pronto vamos a aprender un estimado de locación que es más *robusto* (menos sensible a valores atípicos) que el promedio y la mediana: la media truncada.

La media truncada es un estimado de locación más *robusto* que el promedio y la mediana. Esto significa que es menos sensible a valores atípicos. La media truncada se obtiene de la siguiente manera:

1. Primero ordenamos nuestro conjunto de manera ascendente.
2. Después decidimos qué porcentaje de nuestros datos vamos a truncar. Los valores más comunes suelen variar entre 5% y 25%.
3. Divide el porcentaje acordado entre dos y elimina esa fracción de tus datos del inicio y del final de tu secuencia. Por ejemplo, si decides truncar un 5%, elimina el 2.5% de tus datos del inicio de tu secuencia y el otro 2.5% del final de tu secuencia.
4. Obtén el promedio de los valores restantes.

Afortunadamente, no tenemos que hacer esto manualmente. La librería `scipy` ya ofrece un método para obtener la media truncada fácilmente: